implementations/strategy/StrategyAlchemix.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.2;
+
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./Strategy.sol";
+import "../../interfaces/external/AlcxStakingPools.sol";
+import "../../interfaces/external/MetaCurvePools.sol";
+import "../../interfaces/external/Uniswap.sol";
+
+// owner == harvester
+contract StrategyAlchemix is Strategy, Ownable {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address constant public alcx = 0xdBdb4d16EdA451D0503b854CF79D55697F90c8DF;
+    address constant public sushiRouter = 0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F;
+    address constant public weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2; // used for alcx <> weth <> usdt route
+    StakingPools public alcxStakingPools = StakingPools(0xAB8e74017a8Cc7c15FFcCd726603790d26d7DeCa);
+    address constant public alcx3CrvCurvePool = 0x43b4FdFD4Ff969587185cDB6f0BD875c5Fc83f8c;
+    MetaCurvePools public metaCurvePools = MetaCurvePools(0xA79828DF1850E8a3A3064576f380D90aECDD3359);
+    uint256 constant public alcxCrvPoolId = 4;
+    int128 constant public usdtIndexInCrvMetapool = 3;
+
+    constructor(address _want) public Strategy(_want) {
+    }
+
+    function update(address _newStratrgy) public override {
+        require(msg.sender == governance, "!governance");
+        withdraw(1e18); // withdraw 100%
+        uint256 _balance = IERC20(want).balanceOf(address(this));
+        IERC20(want).safeTransfer(_newStratrgy, _balance);
+        IVaultX(vaultX).setStrategy(_newStratrgy);
+        IVaultY(vaultY).setStrategy(_newStratrgy);
+    }
+
+    function deposit(uint256 _ne18) public override {
+        require(msg.sender == owner() || msg.sender == governance, "!authorized");
+        uint256 _amount = IERC20(want).balanceOf(address(this)).mul(_ne18).div(1e18);
+        IERC20(want).safeApprove(address(metaCurvePools), 0);
+        IERC20(want).safeApprove(address(metaCurvePools), _amount);
+        uint256[4] memory amountsToAdd = [
+            uint256(0),
+            uint256(0),
+            uint256(0),
+            uint256(usdtIndexInCrvMetapool)
```

Comment:
`uint256(usdtIndexInCrvMetapool)` should be `IERC20(want).balanceOf(address(this)` ?